tests/test_sketch.py

```diff
@@ -41,16 +41,123 @@
     Polygon,
     Sketch,
     SketchCircle,
+    SketchEdge,
     SketchEllipse,
     SketchSegment,
     Slot,
+    SpurGear,
+    Triangle,
 )
 
 from .conftest import are_graphics_available
 
 DOUBLE_EPS = np.finfo(float).eps
 
 
+def test_sketch_sketch():
+    """Test the sketch sketch tag functionality"""
+    sketch = Sketch()
+    sketch.segment(Point2D([-4, 5], unit=UNITS.m), Point2D([4, 5], unit=UNITS.m))
+    sketch.tag = "SketchTag"
+    assert sketch.tag == "SketchTag"
+
+
+def test_sketch_segment():
+    """Test the sketch segment unit conversion for end point, not equal,
+    and plane change functionality"""
+    start_point = Point2D([0, 0], unit=UNITS.meter)
+    end_point = Point2D([5, 5], unit=UNITS.kilometer)
+    segment1 = SketchSegment(start_point, end_point)
+    assert segment1.end == Point2D([5000, 5000])
+    segment2 = SketchSegment(start_point, end_point)
+    assert segment1.__ne__(segment2) is False
+    segment2 = SketchSegment(end_point, start_point)
+    assert segment1.__ne__(segment2) is True
+    new_plane = Plane(
+        origin=Point3D([0, 0, 5]), direction_x=Vector3D([1, 0, 0]), direction_y=Vector3D([0, 0, 1])
+    )
+    segment1.plane_change(new_plane)
+    assert segment1.direction == UnitVector3D([0.70710678, 0.0, 0.70710678])
+
+
+def test_sketch_gears():
+    """Test the sketch gears addendum, dedendum, and visualization polydate functionality"""
+    origin = Point2D([0, 1], unit=UNITS.meter)
+    module = 40
+    pressure_angle = Quantity(20, UNITS.deg)
+    n_teeth = 22
+    gearspur = SpurGear(origin, module, pressure_angle, n_teeth)
+    assert gearspur.addendum == 40.0
+    assert gearspur.dedendum == 50.0
+    gearspurpoly = gearspur.visualization_polydata
+    assert gearspurpoly.center == pytest.approx(
+        ([0.0, 1.0000000298023224, 0.0]),
+        rel=1e-7,
+        abs=1e-8,
+    )
+    assert gearspurpoly.bounds == pytest.approx(
+        [-0.4852221608161926, 0.4852221608161926, 0.5148882269859314, 1.4851118326187134, 0.0, 0.0],
+        rel=1e-7,
+        abs=1e-8,
+    )
+    assert gearspurpoly.n_cells == 704
+    assert gearspurpoly.n_points == 70405
+    assert gearspurpoly.n_open_edges == 0.0
+
+
+@pytest.mark.skipif(
+    not are_graphics_available(), reason="Skipping due to graphics requirements missing"
+)
+def test_sketch_circle_plane_change():
```

Comment:
I think you might have skipped the wrong test - sorry! It should be the previous one ``test_sketch_gears``

Reply:
Awesome, thanks @syscordan!